webui/react/src/components/TrialActions.tsx

```diff
@@ -10,20 +10,23 @@ import { terminalRunStates } from 'utils/types';
 
 import css from './TrialActions.module.scss';
 
-interface Props {
-  trial: TrialDetails;
-  onSettled: () => void; // A callback to trigger after an action is done.
-}
-
 enum Action {
   Continue = 'Continue',
   Logs = 'Logs',
   Tensorboard = 'Tensorboard',
 }
 
+interface Props {
+  trial: TrialDetails;
+  onSettled: () => void; // A callback to trigger after an action is done.
+  onClick: {
+    [key in Action]?: () => void;
+  }
```

Comment:
**suggestion:** this is an unusual pattern and bit hard to follow. I recommend just the standard:
```
onClick: () => Action;
```
and let the parent handle what to do with the different `Action`s it receives.